fix(content): libraries section update (#1115)
filecoin-project/specs: content/libraries/multiformats/_index.md

---
title: Multiformats
weight: 4
dashboardWeight: 1
dashboardState: stable
dashboardAudit: missing
dashboardTests: 0
---

> # Multiformats - self describing protocol values

Reviewer: These headers appear in the TOC. I think it's better if they just name the topic rather than try to describe it, which can be done in the body text.

> [Multiformats](https://multiformats.io/) is a set of formating rules and formatting structures for self-describing values. These values are useful both to the data layer (IPLD) and to the network layer (libp2p). Multiformats includes specifications for the Content Identifier (CID) used by IPLD and IPFS, the multicodec, multibase and multiaddress (used by libp2p).

Reviewer 2: I would cut the first sentence down to “Multiformats is a set of self-describing values”

Please refer to the [Multiformats repository](https://github.com/multiformats) for more information.

## CIDs - Content IDentifiers

Filecoin references data using IPLD's Content Identifier (CID).

> A CID is effectively a hash value, prefixed with its hash function (multihash) as well as extra labels, such as a `codec` and a `multibase` to inform applications about how to deserialize the given data.

Reviewer 2: `multibase` isn’t part of CID, and I’m pretty sure we don’t use multibase in Filecoin for anything other than printing the CIDs.

I would maybe pair this down to something like:

“
A CID is a hash digest prefixed with identifiers for its hash function and codec. This means you can validate and decode data by with only this identifier.

When CIDs are printed as strings they also use `multibase` to identify the base encoding being used.
“

---
title: Multiformats
weight: 3
dashboardWeight: 1
dashboardState: stable
dashboardAudit: missing
dashboardTests: 0
---

# Multiformats

[Multiformats](https://multiformats.io/) is a set of self-describing protocol values. These values are useful both to the data layer (IPLD) and to the network layer (libp2p). Multiformats includes specifications for the Content Identifier (CID) used by IPLD and IPFS, the multicodec, multibase and multiaddress (used by libp2p).

Please refer to the [Multiformats repository](https://github.com/multiformats) for more information.

## CIDs

Filecoin references data using IPLD's Content Identifier (CID).

> A CID is a hash digest prefixed with identifiers for its hash function and codec. This means you can validate and decode data by with only this identifier.

Reviewer 3: ```suggestion
A CID is a hash digest prefixed with identifiers for its hash function and codec. This means you can validate and decode data with only this identifier.
```

When CIDs are printed as strings they also use multibase to identify the base encoding being used.

> For a more detailed specification, we refer the reader to the

Reviewer 3: ```suggestion
For a more detailed specification, please see the
```

[CID specification](https://github.com/multiformats/cid).

## Multihash

> A Multihash is a set of self-describing hash values. Multihash is used for differentiating outputs from various well-established cryptographic hash functions, addressing sizes and encoding considerations.

Reviewer 3: ```suggestion
A Multihash is a set of self-describing hash values. Multihash is used for differentiating outputs from various well-established cryptographic hash functions, while addressing size and encoding considerations.
```
Not sure I'm reading this sentence right, so changed it to make it less confusing to me (but please reject if I read it wrong)
